CRM/Member/Form/MembershipRenewal.php

```diff
@@ -493,20 +557,24 @@ protected function submit() {
     $this->assign('receive_date', $this->_params['receive_date']);
     $this->processBillingAddress();
     list($userName) = CRM_Contact_BAO_Contact_Location::getEmailDetails(CRM_Core_Session::singleton()->get('userID'));
-    $this->_params['total_amount'] = CRM_Utils_Array::value('total_amount', $this->_params,
-      CRM_Core_DAO::getFieldValue('CRM_Member_DAO_MembershipType', $this->_memType, 'minimum_fee')
-    );
-    $this->_membershipId = $this->_id;
-    $customFieldsFormatted = CRM_Core_BAO_CustomField::postProcess($this->_params,
```

Comment:
Only calculate total_amount, using minimum_fee for non-price sets renewals. For price sets ones, this will be calculated.
_membershipId will only apply for non-price set ones, as their may be multiple memberships being renewed when using price sets.

Reply:
postProcess moved to submitRenewMembershipSingle